apps/187-navbar-collapse/app.R

```diff
@@ -0,0 +1,41 @@
+library(shiny)
+
+msg <- tags$h6(
+  "If it isn't already, make your window narrow enough so that a menu appears above (and to the right). ",
+  "Clicking that menu should show (and hide) a nav dropdown. ",
+  "Confirm that the nav dropdown can be shown/hidden, and that you can click",
+  "'Summary' to view a data summary (and 'Plot' to see a plot)."
+)
+
+jster <- shinyjster::shinyjster_js(
+  "
+    var jst = jster();
+    jst.add(Jster.shiny.waitUntilStable);
+    jst.add(function() {
+       var toggle = $('.navbar-toggle:visible');
+       var nav    = $('.navbar-collapse:visible');
+       Jster.assert.isEqual(toggle.length, 1, 'Failed to find collapsible menu, does the window need to be resized?');
+       Jster.assert.isEqual(nav.length, 0, 'The collapsible navbar should not be visible by default');
+       toggle.click(function() {
+         Jster.assert.isEqual(nav.length, 1, 'Clicking the navbar toggle should make the navbar appear.');
+       });
+    });
```

Comment:
To capture the asserts, it is best to not mix sync and async tests.  So, we click and move on to the next test once the click has responded.

```suggestion
    var toggle;
    var nav;
    jst.add(function() {
       toggle = $('.navbar-toggle:visible');
       nav    = $('.navbar-collapse:visible');
       
       Jster.assert.isEqual(toggle.length, 1, 'Failed to find collapsible menu, does the window need to be resized?');
       Jster.assert.isEqual(nav.length, 0, 'The collapsible navbar should not be visible by default');
    });

    jst.add(function(done) {
       toggle.click(done);
    });

    jst.add(function() {
      Jster.assert.isEqual(nav.length, 1, 'Clicking the navbar toggle should make the navbar appear.');
    });
```